### ISSUE TITLE
leftwm-theme error for libgit2 >= 1.5

### ISSUE THREAD

**Issue by stevensonmt** [leftwm/leftwm-theme#64]
### Describe the bug

```
$ leftwm-theme --help
leftwm-theme: error while loading shared libraries: libgit2.so.1.4: cannot open shared object file: No such file or directory
```

### Expected behavior (optional)

Should allow libgit2 >= some minimal version rather than being fixed on specific version.

### Steps to reproduce (optional)

Upgrade libgit2 to 1.5.
Run leftwm-theme.

### OS / Distro

Linux 6.1.5-arch2-1

### LeftWM Check

```text
$ leftwm-check
:: LeftWM version: 0.4.1
:: LeftWM git hash: 438b073
:: Enabled features: default journald-log lefthk lefthk-core tracing-journald
:: Checking feature dependencies . . .
    -> journald-log OK
    -> lefthk OK
:: Loading configuration . . .
    -> Configuration loaded OK
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK
:: Checking theme . . .
WARN: TOML as config format is about to be deprecated.
      Please consider migrating to RON or contact the theme creator about this topic.
      Note: make sure the `up` script is loading the correct theme file.
WARN: TOML as config format is about to be deprecated.
      Please consider migrating to RON or contact the theme creator about this topic.
      Note: make sure the `up` script is loading the correct theme file.
ERROR: File not found: theme.ron
```
```

**Comment by VuiMuich** [comment]
I believe this belongs to the `leftwm-theme` repo, actually. @mautamu do you think you could look into this, as you are still more familiar with this one?

**Comment by stevensonmt** [comment]
I think this was a me issue not a leftwm issue.

**Comment by bfg-coding** [comment]
@stevensonmt how did you resolve this?  I'm having the same issue

**Comment by stevensonmt** [comment]
I had somehow gotten libgit2 ahead of the Arch extra repo (I had temporarily enabled testing at one point and forgot). I just downgraded libgit2 back to the version 1.5 in extra. I haven't tried to build leftwm since the repo version went to 1.6 though. Looks like the issue is the git2 version in Cargo.toml is fixed on 0.15. `git2 = {version = "0.17"}` would probably fix the issue so I'm reopening the issue.

**Comment by stevensonmt** [comment]
I think this got reintroduced. I'm getting the same error with leftwm-theme now with 0.5.1-2 installed from AUR package. Looks like git2 crate is up to 0.18. Not able to check if that fixes the issue just now.

I wonder if adding `vendored-libgit2` to required features in `cargo.toml` would future proof this at all. For now I've just done a probably dumb thing and symlinked the current `libgit2.so` to `libgit2.so.1.4`. Seems to work as far as leftwm-theme is concerned.
